src/ucs/stats/stats.c

```diff
@@ -296,6 +280,27 @@ static void ucs_stats_add_to_filter(ucs_stats_node_t *node,
             temp_filter_node = temp_filter_node->parent;
         }
     }
+
+    return UCS_OK;
+}
+
+static int ucs_stats_node_add(ucs_stats_node_t *node, ucs_stats_node_t *parent)
+{
+    ucs_status_t status;
+    ucs_assert(node != &ucs_stats_context.root_node);
+    if (parent == NULL) {
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    /* Append node to existing tree */
+    pthread_mutex_lock(&ucs_stats_context.lock);
+    ucs_list_add_tail(&parent->children[UCS_STATS_ACTIVE_CHILDREN], &node->list);
+    node->parent = parent;
+
+    status = ucs_stats_add_to_filter(node, &ucs_stats_context.root_filter_node);
```

Comment:
ucs_stats_add_to_filter may fail only on malloc failure. So there is nothing to clean. when returning to the caller, node is being free.